### ISSUE TITLE
Add a generic IndexSpider class

### ISSUE THREAD

**Issue by yolile** [open-contracting/kingfisher-collect#337]
The Honduras API returns a json like:

``` json
  "releases": 991869,
  "pages": 99187,
  "page": "1",
  "next": "http://www.contratacionesabiertas.gob.hn/api/v1/record/?format=json&page=2",
  "previous": null,
  "recordPackage": {...}
```
We can use the `pages` field to build the next url in a loop and make the request in parallel

**Comment by jpmckinney** [comment]
Let's also check the other spiders that inherit from `LinksSpider`, in case any return the total count of pages/results.

In doing so, we can create a new superclass for all spiders that can generate requests based on a total count (maybe `IndexSpider`). These include at least:

* [ ] kenya_makueni
* [ ] mexico_administracion_publica_federal
* [ ] mexico_quien_es_quien
* [ ] montreal
* [ ] uganda_releases
* [ ] uk_contracts_finder

We can add a note to either the docstring or `writing-spiders.rst` to say that we prefer using `IndexSpider` instead of `LinksSpider` where possible, since generating all requests at once will lead to a faster collection.

**Comment by romifz** [comment]
Closed by #497
